File: fixtures/java/controller-service-provider/src/main/java/chakra/payments/shared/Shared.java
package chakra.payments.shared;

/** Shared targets: one high fan-in-style static callee and one uniquely named callee. */
public final class Shared {

    private Shared() {
    }

    public static void recordEvent(String kind) {
        if (kind.isEmpty()) {
            throw new IllegalArgumentException("event kind must not be empty");
        }
    }

    public static void sharedUniqueTarget() {
        // deliberately empty: the callers scenario asserts exactly one caller
    }
}

File: fixtures/csharp/controller-service-provider/src/Payments.Core/Shared.cs
namespace Chakra.Payments.Shared;

public static class Shared
{
    public static string NormalizePayment(this string value)
    {
        return value.Trim();
    }

    public static void RecordEvent(string name)
    {
        Console.WriteLine(name);
    }

    public static string SharedUniqueTarget(string value)
    {
        return value.Trim();
    }
}
